docs/docs/tutorials/rag/index.ipynb

```diff
@@ -191,78 +293,38 @@
-    }
-   ],
-   "source": [
-    "cot = dspy.ChainOfThought('question -> response')\n",
-    "cot(question=\"should curly braces appear on their own line?\")"
+    "dspy.inspect_history()"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "## Using a DSPy Optimizer to Improve Your RAG Prompt\n",
     "\n",
-    "\n",
-    "Interestingly, asking for reasoning can make the output `response` shorter in this case. Is this a good thing or a bad thing? It depends on what you need: there's no free lunch, but DSPy gives you the tools to experiment with different strategies extremely quickly.\n",
-    "\n",
-    "By the way, `dspy.ChainOfThought` is implemented in DSPy, using `dspy.Predict`. This is a good place to `dspy.inspect_history` if you're curious.\n"
+    "You can stop reading if you just want to build a RAG application with DSPy, in this section we are going to show how to use DSPy optimizer to improve the program's quality."
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Using DSPy well involves evaluation and iterative development.\n",
-    "\n",
-    "You already know a lot about DSPy at this point. If all you want is quick scripting, this much of DSPy already enables a lot. Sprinkling DSPy signatures and modules into your Python control flow is a pretty ergonomic way to just get stuff done with LMs.\n",
-    "\n",
-    "That said, you're likely here because you want to build a high-quality system and improve it over time. The way to do that in DSPy is to iterate fast by evaluating the quality of your system and using DSPy's powerful tools, e.g. Optimizers.\n",
-    "\n",
-    "## Manipulating Examples in DSPy.\n",
-    "\n",
-    "To measure the quality of your DSPy system, you need (1) a bunch of input values, like `question`s for example, and (2) a `metric` that can score the quality of an output from your system. Metrics vary widely. Some metrics need ground-truth labels of ideal outputs, e.g. for classification or question answering. Other metrics are self-supervised, e.g. checking faithfulness or lack of hallucination, perhaps using a DSPy program as a judge of these qualities.\n",
-    "\n",
-    "Let's load a dataset of questions and their (pretty long) gold answers. Since we started this notebook with the goal of building **a system for answering Tech questions**, we obtained a bunch of StackExchange-based questions and their correct answers from the [RAG-QA Arena](https://arxiv.org/abs/2407.13998) dataset.\n",
-    "\n"
+    "Let's first evaluate the RAG we built. In order to evaluate a RAG application, we need a dataset along with an evaluation metric. For this purpose, we will use the [RAG-QA dataset](https://arxiv.org/abs/2407.13998) as the evaluation dataset, and [SemanticF1](https://karmake2.github.io/files/Publications/2022/SEM_F1.pdf) as the evaluation metric. Let's download the dataset and import the evaluation metric."
```

Comment:
just comment: I like the DSPy tutorials that are research/science oriented. But we should probably provide a separate doc for non-DS or non-Eng backgroup people 😄